src/meta/processors/parts/CreateSpaceAsProcessor.cpp

```diff
@@ -0,0 +1,221 @@
+/* Copyright (c) 2018 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/processors/parts/CreateSpaceAsProcessor.h"
+
+#include "meta/ActiveHostsMan.h"
+
+namespace nebula {
+namespace meta {
+
+void CreateSpaceAsProcessor::process(const cpp2::CreateSpaceAsReq &req) {
+  SCOPE_EXIT {
+    if (rc_ != nebula::cpp2::ErrorCode::SUCCEEDED) {
+      handleErrorCode(rc_);
+    }
+    onFinished();
+  };
+
+  folly::SharedMutex::WriteHolder wHolder(LockUtils::spaceLock());
+  auto oldSpaceName = req.get_old_space_name();
+  auto newSpaceName = req.get_new_space_name();
+  auto oldSpaceId = getSpaceId(oldSpaceName);
+  auto newSpaceId = getSpaceId(newSpaceName);
```

Comment:
`if (nebula::ok(newSpaceId)) {
    auto ret = nebula::cpp2::ErrorCode::SUCCEEDED;
    if (!req.get_if_not_exists()) {
      LOG(ERROR) ...
      ret = nebula::cpp2::ErrorCode::E_EXISTED;
    }
    resp_.set_id(...);
    handleErrorCode(ret);
    onFinished();
    return;
  }`

Reply:
Maybe we should not `support if_not_exists` in this sentence?

Reply:
When newSpaceName exists, an error should be reported directly

Reply:
> OK, Maybe we should not `support if_not_exists` in this sentence?

Yes, I agree not to support if_not_exists. If so, we need to delete this parameter in the relevant place, such as thrift interface.